### ISSUE TITLE
TypeError: Cannot read property '$ns' of undefined

### ISSUE THREAD

**Issue by ottosson** [peerlibrary/node-xml4js#8]
When I try to parse a xml file I get the error "TypeError: Cannot read property '$ns' of undefined", and it seems to be coming from xml2js. I just copied and pasted the example code:

    xml4js.parseString(xml, {downloadSchemas: true}, function (err, result) {
        console.log(util.inspect(result, false, null));
    });

I use xml2js 0.4.9 and xml4js 0.1.2.

**Comment by mitar** [comment]
Sorry, I will not be able to help you much here without more data. Can you do a pull request with a failing test case?

**Comment by thewindev** [comment]
Hi! I'm facing the same issue. Here's an XML example:
<?xml version='1.0' encoding='UTF-8'?><entry xmlns='http://www.w3.org/2005/Atom' xmlns:gsa='http://schemas.google.com/gsa/2007'><id>http://213.229.151.228:8002/feeds/status/documentStatus</id><updated>2015-10-15T08:33:58.161Z</updated><app:edited xmlns:app='http://purl.org/atom/app#'>2015-10-15T08:33:58.161Z</app:edited><link rel='self' type='application/atom+xml' href='https://213.229.151.228:8002/feeds/status/documentStatus'/><link rel='edit' type='application/atom+xml' href='https://213.229.151.228:8002/feeds/status/documentStatus'/><gsa:content name='entryID'>documentStatus</gsa:content><gsa:content name='servedURLs'>1,180</gsa:content><gsa:content name='crawlPagesPerSecond'>0.03</gsa:content><gsa:content name='crawledURLsToday'>0</gsa:content><gsa:content name='foundURLs'>96,362</gsa:content><gsa:content name='filteredBytes'>208,478,568,448</gsa:content><gsa:content name='errorURLsToday'>0</gsa:content></entry>

**Comment by mlooney** [comment]
Same issue here:


```
<AmazonEnvelope xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="amzn-envelope.xsd">

```

xmllint has no problem, and validates the document successfully.  Note that amzn-envelope.xsd includes all sorts of other xsds as well

**Comment by mitar** [comment]
Yes, as I mentioned in other issues, I made this package for a particular XSD I had and developed features only for what was needed. I know some things are not implemented. But at the moment I do not have time to further work on this, especially because it works for me. But I would be glad to accept pull requests adding more features.

**Comment by mitar** [comment]
One thing, this package uses a fork of xml2js because two pull requests have not been merged upstream. Are you using that fork or not?

**Comment by mitar** [comment]
Can people create pull requests with tests which are failing?

**Comment by mitar** [comment]
OK, and people, those are different issues here. Do not all write in the same ticket. Are you all getting `Cannot read property '$ns' of undefined`? You are not.

@thewindev: You problem is that for `http://www.w3.org/2005/Atom` namespace there is no XML Schema definition (XSD) location provided.

@mlooney: Your example is not a valid XML?

**Comment by mlooney** [comment]
no, that particular snippet is not, but the document is rather large, so i didn't paste it in its entirety. I'll come up with a smaller test case to reproduce.

**Comment by mlooney** [comment]
```
<?xml version="1.0"?>
<Element xmlns:xsi="x" xsi:noNamespaceSchemaLocation="y">
</Element>
```

This fails _before_ validation as well. xmllint has no trouble (it will fail validation, but that's expected).  I suspect the lib is assuming that all elements will be namespaced (adding a ns to the root element causes an actual validation error, as expected). So i guess it boils down to not supporting noNamespaceSchemaLocation.

**Comment by mlooney** [comment]
demo:

https://gist.github.com/mlooney/b9c489ee01b7f3fd3102f2efa31313a9

**Comment by mitar** [comment]
I see, `noNamespaceSchemaLocation` seems to not be supported. That should be a fairly straightforward addition here: https://github.com/peerlibrary/node-xml4js/blob/master/lib/xsd.js#L515
